### ISSUE TITLE
Links within snackbars have insufficient color contrast ratio on hover

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#47273]
### Description

When hovering a link within a Snackbar, the link hover style changes the link color to a blue `#007cba`. This color has insufficient contrast ratio against the background color `#1e1e1e`. The contrast ratio is only 3.65:1. It should be at least 4.5:1.

<img width="302" alt="Screenshot 2023-01-19 at 09 58 01" src="https://user-images.githubusercontent.com/1682452/213402029-03b39db1-e816-4976-bcbb-d8867c5c0916.png">

### Step-by-step reproduction instructions

- Edit a draft post and save it as draft.
- Wait for the Snackbar to appear at the bottom left of the screen.
- Hover the 'View Preview' link.
- Observe the link color changes to blue `#007cba`.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by MahendraBishnoi29** [comment]
hey @afercia i would love to work on this.

**Comment by afercia** [comment]
@MahendraBishnoi29 Sure, you're welcome to submit a PR! If you're new to contributing, you may want to have a look at https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/README.md

**Comment by MahendraBishnoi29** [comment]
also @afercia can you please guide me on how do I fix this, I mean where do I make these changes I'm just new here.

**Comment by afercia** [comment]
I think the css (actually scss( of the Snackbar component is `packages/components/src/snackbar/style.scss`

**Comment by codeofjan** [comment]
@afercia what color with sufficient contrast would you suggest?

**Comment by MahendraBishnoi29** [comment]
@afercia  here the `$components-color-accent` is the color when we hover on the snackbar link right? if yes then should i update the color directly here or somewhere else?

```
.components-snackbar {
	font-family: $default-font;
	font-size: $default-font-size;
	background-color: $gray-900;
	border-radius: $radius-block-ui;
	box-shadow: 0 2px 4px rgba(0, 0, 0, 0.3);
	color: $white;
	padding: 16px 24px;
	width: 100%;
	max-width: 600px;
	box-sizing: border-box;
	cursor: pointer;

	pointer-events: auto;

	@include break-small() {
		width: fit-content;
	}

	&:focus {
		box-shadow:
			0 0 0 1px $white,
			0 0 0 3px $components-color-accent;
	}
}

.components-snackbar__action.components-button {
	margin-left: $grid-unit-40;
	color: $white;
	height: auto;
	flex-shrink: 0;
	line-height: $default-line-height;
	padding: 0;

	&:not(:disabled):not([aria-disabled="true"]):not(.is-secondary) {
		text-decoration: underline;
		background-color: transparent;

		&:focus {
			color: $white;
			box-shadow: none;
			outline: 1px dotted $white;
		}

		&:hover {
			color: $components-color-accent; //  <------ this
		}
	}
}
```

**Comment by afercia** [comment]
@codeofjan @MahendraBishnoi29 thanks for looking into this.
Yes, looks like that's the `$components-color-accent;` is the color applied on hover. However, I wonder whether there's the need of a hover style in the first place. To me, the pointer that changes to `cursor` is a sufficient indication.
I think there's the need for some design feedback here.
If a hover style is preferred, I'd suggest to not use a color change. Instead, I'd recommend a _shape_change. For example, removing the underline on hover.

**Comment by MahendraBishnoi29** [comment]
@afercia so should I only remove the underline on hover? that's it?

**Comment by afercia** [comment]
Let's wait form some design feedback 🙂

**Comment by MahendraBishnoi29** [comment]
sure😃 @afercia

**Comment by codeofjan** [comment]
Any updates for design feedback here?

**Comment by t-hamano** [comment]
> If a hover style is preferred, I'd suggest to not use a color change. Instead, I'd recommend a _shape_change. For example, removing the underline on hover.

I have tested this approach using the browser developer tools. It seems natural to me 🙂


https://user-images.githubusercontent.com/54422211/230769136-71fb8bc5-7258-49cf-9a4b-9f4f8ab1170b.mp4

**Comment by MahendraBishnoi29** [comment]
> > If a hover style is preferred, I'd suggest to not use a color change. Instead, I'd recommend a _shape_change. For example, removing the underline on hover.
> 
> I have tested this approach using the browser developer tools. It seems natural to me 🙂
> 
> 
> https://user-images.githubusercontent.com/54422211/230769136-71fb8bc5-7258-49cf-9a4b-9f4f8ab1170b.mp4
> 
> 
> 

where did you change the color? i mean could you please highlight the code in my comment above 🙂

**Comment by t-hamano** [comment]
@MahendraBishnoi29 

In the code base, the changes may be as follows

```diff
diff --git a/packages/components/src/snackbar/style.scss b/packages/components/src/snackbar/style.scss
index 0e54d3b432..9e9acebb63 100644
--- a/packages/components/src/snackbar/style.scss
+++ b/packages/components/src/snackbar/style.scss
@@ -64,7 +64,8 @@
                }
 
                &:hover {
-                       color: $components-color-accent;
+                       text-decoration: none;
+                       color: $white;
                }
        }
 }
```

Also, by executing the following code in the browser console, you can display a button that does not disappear naturally, making it easier to debug:

```javascript
wp.data.dispatch( 'core/notices' ).createSuccessNotice(
	'Post updated.',
	{
		type: 'snackbar',
		isDismissible: false,
		explicitDismiss: true,
		actions: [
			{
				url: '#',
				label: 'View post',
			},
		],
	}
);
```

@jasmussen 

In #17887 regarding the snack bar link, I think there was no specific mention of blue text on hover (It's been a few PR years now 😅). What do you think about the approach of not changing the text color on hover, but instead removing the underline?

**Comment by jasmussen** [comment]
Seems an unfortunate oversight that the color on hover had insufficient contrast, and yes indeed, I don't think it needs very much in the way of hover decoration. Underlined text does seem fine, can even do without that. An alternative if we do want to keep the color change, is to change the background color of the clickable button, to be black for example.

**Comment by MahendraBishnoi29** [comment]
@jasmussen @t-hamano so should I create a PR for this issue now? I'll be adding the `underline` on hover with the color `white`

**Comment by jasmussen** [comment]
Go for it, with or without hover underline, it's good to fix this. I think snackbars are worth revisiting visually as a whole, but we can do that separately.

**Comment by ciampo** [comment]
Closing this issue since https://github.com/WordPress/gutenberg/pull/49682 got merged — although feel free to reopen if necessary
